controller/grpc.go

```diff
@@ -951,6 +952,52 @@ func (g *grpcAPI) SetRoutes(ctx context.Context, req *api.SetRoutesRequest) (*ap
 	}, nil
 }
 
+func (g *grpcAPI) ListAppRoutes(ctx context.Context, req *api.ListAppRoutesRequest) (*api.ListAppRoutesResponse, error) {
+	if len(req.Apps) == 0 {
+		return nil, hh.ValidationErr("apps", "must be set")
```

Comment:
Should this and the other errors returned be `grpc.Error`s?

Reply:
Errors refactored in aa4ceb32.